### PR TITLE
Add "Declarative extension style" to declarative.md

### PR DESCRIPTION
**Pull request by lukepighetti**
_Description of what this PR is changing or adding, and why:_

There is some confusion in the community over the declarative nature of the Widget extension pattern that is becoming more commonly used. Folks are pointing to this article saying that they are imperative because they are using method calls.

_Issues fixed by this PR (if any):_

Confusion in the community

_How reviewers can help:_

I would appreciate suggestions to improve the technical writing

_Notes:_

I did not reuse the ViewB/ViewC example because it's not a pattern that I recommend for extension Widget composition and in my experience people will use this as an indication of patterns to follow

_Community context:_

- https://twitter.com/luke_pighetti/status/1667909336088170498
- https://twitter.com/_incendial/status/1667911906936455169
- https://twitter.com/BrandoTrautmann/status/1667914597108535296
- https://twitter.com/TheKnaecke/status/1667906454295117824
- https://twitter.com/caseycrogers/status/1667894024848674817

_Industry context:_

- https://developer.apple.com/xcode/swiftui/ "Declarative syntax" section

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by SashaKryzh** [comment]
I would suggest using a move commonly used widget for an example (e.g. `Padding`), because it would make more sense to beginners.

**Comment by lukepighetti** [comment]
@SashaKryzh Should I adjust all the examples to use padding? 🤔

**Comment by SashaKryzh** [comment]
@lukepighetti I am not sure what you mean by all examples, because I see only 1 file changed.

**Comment by lukepighetti** [comment]
I switched to using `Center()` and `.center()` in the example

File: src/get-started/flutter-for/declarative.md [hunk 1]
```diff
@@ -68,3 +68,27 @@ RenderObjects persist between frames and Flutter’s lightweight Widgets
...
 
+## Declarative extension style
+
+There is a new declarative pattern that has emerged as a result of Dart 
+adding the `extension` language feature. While this pattern is not currently 
+in use in the Flutter framework, it is actively in use within the community.
+
+This declarative style does appear at first glance to be similar to the imperative
+style shown above, but the underlying operation is composing widgets through
+their constructor just like in the declarative style shown above.
```
Comment (atsansone) [inline src/get-started/flutter-for/declarative.md:79]: **issues:**
- Write in active voice per [GSG](https://developers.google.com/style/voice).
- Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).

```suggestion
The Flutter community surfaced a new declarative pattern as a result of Dart 
adding the `extension` language feature. The Flutter framework doesn't use
this pattern at this time. The community does use this pattern.

This declarative style does appear to resemble the imperative style that
other frameworks use. However, the underlying operation composes
widgets through their constructor just like in the [declarative style](#why-a-declarative-ui).
```

**Review by atsansone — CHANGES_REQUESTED** [review]
@lukepighetti : Thanks for your contribution! Some style issues needed updating. If you accept these changes, we can approve and merge.

File: src/get-started/flutter-for/declarative.md (hunk 1 above)
Comment (munificent) [reply src/get-started/flutter-for/declarative.md:79]: As far as general Dart style is concerned, sure, there's nothing wrong with using extensions. As to whether that's good style in Flutter code and/or worth documenting it on Flutter's own docs... that's outside of my area of expertise.

I do find this section confusing. It's responding to an argument that isn't stated on the doc itself. Anyone who comes to this page who *hasn't* already heard someone claim that extension methods aren't declarative will have little context as to why the doc is justifying them and be confused as to what it's trying to get at.

Also, this section emphasizes extension methods, but it's not the fact that these are *extension* methods that makes them declarative. You can have imperative extension methods and declarative instance methods:

```dart
extension ListExt<T> on List<T> {
  void imperative(T t) {
    add(t);
    add(t);
  }
}

class MyList<T> extends ListBase<T> {
  List<T> declarative(T t) => [...this, t, t];
}
```

What matters is whether the method is *mutating* versus creating a new object.

**Comment by sfshaza2** [comment]
Thanks, @munificent, for clarifying! I am closing this PR. @lukepighetti, if you really want to add some text (I know this can be a controversial subject in the twitter-verse), then please address Bob's issues and open a new PR.

**Pull request closed without merging**